packages/components/src/FilterBar/FilterBar.component.js

```diff
@@ -101,143 +100,147 @@ FilterInput.propTypes = {
 	'data-test': PropTypes.string,
 	'data-feature': PropTypes.string,
 	disabled: PropTypes.bool,
-	t: PropTypes.func.isRequired,
 };
 
 /**
  * @param {object} props react props
  * @example
  <FilterBar id="my-filter" docked="false" onFilter="filter()"></Filter>
  */
-export class FilterBarComponent extends React.Component {
-	constructor(props) {
-		super(props);
-		this.onFocus = this.onFocus.bind(this);
-		this.onBlur = this.onBlur.bind(this);
-		this.onFilter = this.onFilter.bind(this);
-		this.onSubmit = this.onSubmit.bind(this);
-		this.onClear = this.onClear.bind(this);
-		this.state = { focus: this.props.focus, value: this.props.value };
-	}
+export default function FilterBar(props) {
+	const { onFocus, onBlur, onToggle, onFilter } = props;
+	const [focus, setFocus] = useState();
+	const [value, setValue] = useState();
+	const { t } = useTranslation(I18N_DOMAIN_COMPONENTS);
 
-	componentWillReceiveProps(nextProps) {
-		if (nextProps.value !== this.state.value) {
-			this.setState({ value: nextProps.value });
-		}
-	}
+	useEffect(() => {
+		// in controlled mode, if the value changes, replace the current value
+		setValue(props.value);
+	}, [props.value]);
 
-	onFocus(event) {
-		this.setState({ focus: true });
-		if (this.props.onFocus) {
-			this.props.onFocus(event);
-		}
-	}
+	const onFocusCallback = useCallback(
+		event => {
+			setFocus(true);
+			if (onFocus) {
+				onFocus(event);
+			}
+		},
+		[onFocus, setFocus],
+	);
 
-	onBlur(event) {
-		this.setState({ focus: false });
-		if (this.props.onBlur) {
-			this.props.onBlur(event);
-		}
-		if (!this.state.value && this.props.onToggle) {
-			this.props.onToggle(event);
-		}
-	}
+	const onBlurCallback = useCallback(
+		event => {
+			setFocus(false);
+			if (onBlur) {
+				onBlur(event);
+			}
+			if (!value && onToggle) {
+				onToggle(event);
+			}
+		},
+		[onBlur, onToggle, setFocus, value],
+	);
```

Comment:
why memoizing this? it will be reallocated on each input anyway...

Reply:
I missed this controlled aspect and expected that on user type event, the input was not rerendered, handling the event value.
But you're right, the input is controlled so at each change it will rerender. I'll remove that